pwiz_tools/Skyline/TestUtil/TestFunctional.cs

```diff
@@ -384,23 +384,51 @@ private static void SkylineInvoke(Action act)
         {
             if (null != SkylineWindow)
             {
-                SkylineWindow.Invoke(act);
+
+                if (SkylineWindow.InvokeRequired)
+                {
+                    SkylineWindow.Invoke(act);
+                }
+                else
+                {
+                    Console.WriteLine(@"# SkylineInvoke() called on UI thread (SkylineWindow)"); // CONSIDER:(bspratt) remove before merge?
+                    act();
+                }
             }
             else
             {
-                FindOpenForm<StartPage>().Invoke(act);
+                var startPage = FindOpenForm<StartPage>();
+
+                if (startPage.InvokeRequired)
+                {
+                    startPage.Invoke(act);
+                }
+                else
+                {
+                    Console.WriteLine(@"# SkylineInvoke() called on UI thread (StartPage)"); // CONSIDER:(bspratt) remove before merge?
+                    act();
+                }
             }
         }
 
         private static void SkylineBeginInvoke(Action act)
         {
             if (null != SkylineWindow)
             {
+                if (!SkylineWindow.InvokeRequired)
```

Comment:
Function prefered to keep the code in each block one line.

Reply:
OK, I can easily simplify.